app/services/pieces_justificatives_service.rb

```diff
@@ -48,10 +42,21 @@ def self.serialize_champs_as_pjs(dossier)
 
   private
 
-  def self.champs_pieces_justificatives_with_attachments(champs)
-    champs
+  def self.pjs_for_champs(dossier, requesting_account)
+    allowed_champs = Pundit.policy_scope(requesting_account, Champ).where(dossier: dossier)
```

Comment:
Maintenant on récupère les champs avec un [gros scope Pundit velu](https://github.com/betagouv/demarches-simplifiees.fr/blob/add-private-pj-to-export/app/policies/champ_policy.rb#L3-L37), qui fait le check sur les instructeurs rattaché à chaque champ, et tout.

Ça peut potentiellement causer un problème de performances. Cette méthode est appelée dans les templates pour déterminer si on doit afficher le menu "Télécharger l'export des PJs" ou pas, donc une ou deux fois par template.

MAIS on scope très vite cette requête mastoc sur `.where(dossier: dossier)`, ce qui devrait grandement limiter le nombre de lignes concernées par la requête. 

@LeSim tu crois que ça peut causer un problème de perf, ou ça va ?

Reply:
Finalement on n'a pas besoin du scope, donc la question ne se pose pas.